File: Objc代理/Objc代理/ViewController.m
//
//  ViewController.m
//  Objc代理
//
//  Created by BaiYing on 2017/3/6.
//  Copyright © 2017年 BY. All rights reserved.
//

#import "ViewController.h"
#import "SubViewController.h"


@interface ViewController () <SubViewDelegate>

@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (IBAction)goSubView:(UIButton *)sender {
    
    [self performSegueWithIdentifier:@"goSub" sender:nil];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.identifier isEqualToString:@"goSub"]) {
        SubViewController *subVC = segue.destinationViewController;
        subVC.delegate = self;
    }
}

- (void)backWithStr:(NSString *)str {
    
    self.label.text = str;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    int a = 10;
    
    printf("%d", a);
}




@end

File: Swift代理/Swift代理/ViewController.swift
//
//  ViewController.swift
//  Swift代理
//
//  Created by BaiYing on 2017/3/6.
//  Copyright © 2017年 BY. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var textF: UITextField!
    
    @IBAction func goSub(_ sender: Any) {
        performSegue(withIdentifier: "goSub", sender: nil)
    }
    
    /// storyboard 跳转处理
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "goSub" {
            print("hi")
            let subVC = segue.destination as! SubViewController
            subVC.delegate = self
            
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }


}

/// 使用扩展继承协议 实现协议方法 可以分离代码
extension ViewController: SubViewDelegate{
    /// 实现代理方法
    func backStr(str: String) {
        self.textF.text = str
    }
}

